banzai/photometry.py

```diff
@@ -23,95 +26,125 @@ def radius_of_contour(contour, source):
-                return image
-
-            # Convert the detections into a table
-            sources = Table(sources)
-
-            # We remove anything with a detection flag >= 8
-            # This includes memory overflows and objects that are too close the edge
-            sources = sources[sources['flag'] < 8]
+            segmentation_map = detect_sources(convolved_data, self.threshold, npixels=self.min_area)
+
+            logger.info('Deblending sources', image=image)
+            # Note that nlevels here is DEBLEND_NTHRESH in source extractor which is 32 by default
+            deblended_seg_map = deblend_sources(convolved_data, segmentation_map,
+                                                npixels=self.min_area, nlevels=32,
+                                                contrast=0.005, progress_bar=False,
+                                                nproc=1)
+            # Convert the segmentation map to a source catalog
+            catalog = SourceCatalog(data, deblended_seg_map, convolved_data=convolved_data, error=error,
+                                    background=bkg.background)
+
+            sources = Table({'x': catalog.xcentroid + 1.0, 'y': catalog.ycentroid + 1.0,
+                             'xwin': catalog.xcentroid_win + 1.0, 'ywin': catalog.ycentroid_win + 1.0,
+                             'xpeak': catalog.maxval_xindex + 1, 'ypeak': catalog.maxval_yindex + 1,
+                             'peak': catalog.max_value,
+                             'a': catalog.semimajor_sigma.value, 'b': catalog.semiminor_sigma.value,
+                             'theta': catalog.orientation.to('deg').value, 'ellipticity': catalog.ellipticity.value,
+                             'kronrad': catalog.kron_radius.value,
+                             'flux': catalog.kron_flux, 'fluxerr': catalog,
+                             'x2': catalog.covar_sigx2.value, 'y2': catalog.covar_sigy2.value,
+                             'xy': catalog.covar_sigxy.value,
+                             'background': catalog.background_mean})
+
+            for r in range(1, 7):
+                radius_arcsec = r / image.pixel_scale
+                sources[f'fluxaper{r}'], sources[f'fluxerr{r}'] = catalog.circular_photometry(radius_arcsec)
+
+            for r in [0.25, 0.5, 0.75]:
+                sources['fluxrad' + f'{r:.2f}'.lstrip("0.")] = catalog.fluxfrac_radius(r)
+
+            sources['flag'] = 0
+
+            # Flag = 1 for sources with bad pixels
+            flag_sources(sources, catalog.labels, deblended_seg_map, image.mask, flag=1, mask_value=1)
+            # Flag = 2 for sources that are deblended
+            flag_deblended(sources, catalog, segmentation_map, deblended_seg_map, flag_value=2)
+            # Flag = 4 for sources that have saturated pixels
+            flag_sources(sources, catalog.labels, deblended_seg_map, image.mask, flag=4, mask_value=2)
+            # Flag = 8 if kron aperture falls off the image
```

Comment:
should `flag_edge_sources` be called here?

Reply:
Good catch